pkg/trie/triedb/proof/util_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package proof
+
+import (
+	"bytes"
+
+	"github.com/ChainSafe/gossamer/internal/database"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/trie/db"
+)
+
+// MemoryDB is an in-memory implementation of the Database interface backed by a
+// map. It uses blake2b as hashing algorithm
+type MemoryDB struct {
+	data           map[common.Hash][]byte
+	hashedNullNode common.Hash
+	nullNodeData   []byte
+}
+
+func memoryDBFromNullNode(nullKey, nullNodeData []byte) *MemoryDB {
```

Comment:
If this is only used once and its the default behavior, could we not just put this in `NewMemoryDB` and remove this helper?

Reply:
True! changed!